src/mapper/java/org/codehaus/jackson/map/jsontype/impl/SubTypeValidator.java

```diff
@@ -78,7 +130,10 @@ public void validateSubType(JavaType type) throws JsonMappingException
 
             // 18-Dec-2017, tatu: As per [databind#1855], need bit more sophisticated handling
             //    for some Spring framework types
-            if (full.startsWith(PREFIX_STRING)) {
+            // 05-Jan-2017, tatu: ... also, only applies to classes, not interfaces
+            if (raw.isInterface()) {
+                ;
```

Comment:
why the semicolon here?

Reply:
Because it's an empty `if` clause. I guess the semicolon is to emphasise this fact.

Reply:
instead of an else if could you not write it as :
`if (!raw.isInterface && full.startsWith(PREFIX_STRING)) {`
?

Reply:
I know, I know... The entire `do..while` with the label on top is atrocious. I'm just backporting from the original repo.
I keep the changes to minimum, so the future backports will have to deal with smaller diff's.